packages/frontend/core/src/components/blocksuite/block-suite-page-list/utils.tsx

```diff
@@ -61,15 +63,43 @@ export const usePageHelper = (blockSuiteWorkspace: BlockSuiteWorkspace) => {
     showImportModal({ workspace: blockSuiteWorkspace, onSuccess });
   }, [blockSuiteWorkspace, openPage, jumpToSubPath]);
 
+  const createLinkedPageAndOpen = useAsyncCallback(
+    async (pageId: string) => {
+      const page = createPageAndOpen();
+      await page.load();
+      const parentPage = blockSuiteWorkspace.getPage(pageId);
+      if (parentPage) {
+        await parentPage.load();
+        const text = parentPage.Text.fromDelta([
+          {
+            insert: ' ',
+            attributes: {
+              reference: {
+                type: 'LinkedPage',
+                pageId: page.id,
+              },
+            },
+          },
+        ]);
+        const [frame] = parentPage.getBlockByFlavour('affine:note');
+        frame && parentPage.addBlock('affine:paragraph', { text }, frame.id);
+        setPageMeta(page.id, {});
+      }
+    },
+    [blockSuiteWorkspace, createPageAndOpen, setPageMeta]
+  );
```

Comment:
Please move this function to a helper method. Maybe into `usePageHelper`?

Reply:
Hmm, this function seems to be in `usePageHelper` now?